File: math/factorial/Rust/factorial.rs
use std::io;

fn fact(n:i32) -> f64 {
    match n {
        0 => 1.0,
        _ => n as f64 * fact(n-1)
    }
}

fn main() {
    println!("Enter a number");
    let mut num = String::new();
    io::stdin().read_line(&mut num);
    let num : i32 = num.trim().parse().unwrap();
    println!("factorial of {} = {}",num,fact(num));
}

File: math/factorial/dart/factorial.dart
void main() {
  print('!1 = ${factorial(1)}');
  print('!2 = ${factorial(2)}');
  print('!3 = ${factorial(3)}');
  print('!4 = ${factorial(4)}');
  print('!5 = ${factorial(5)}');
  print('!6 = ${factorial(6)}');
  print('!7 = ${factorial(7)}');
}

int factorial (int n) => n > 1 ? n * factorial(n-1) : 1;